lib/pharos/phases/configure_master.rb

```diff
@@ -90,16 +87,22 @@ def generate_config
         config = {
           'apiVersion' => 'kubeadm.k8s.io/v1alpha1',
           'kind' => 'MasterConfiguration',
+          'nodeName' => @host.hostname,
           'kubernetesVersion' => Pharos::KUBE_VERSION,
-          'apiServerCertSANs' => [@host.address, @host.private_address].compact.uniq,
+          'api' => {
+            'advertiseAddress' => @host.peer_address,
+            'controlPlaneEndpoint' => '127.0.0.1'
```

Comment:
I thought the `ControlPlaneEndpoint` was validated to be a DNS address, I'm a little surprised that an IP address also works?

Reply:
https://github.com/kubernetes/kubernetes/blob/v1.10.1/cmd/kubeadm/app/util/endpoint.go#L44

Reply:
https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L132-L143

Reply:
Yeah, it seems almost accidential that IPv4 addresses also pass that validation... I don't think that's intentional, and e.g. IPv6 addresses won't.

Reply:
I'd consider changing this to `localhost` (although check for `::1` vs `127.0.0.1`)

Reply:
Changed to `localhost`.